corehq/apps/cloudcare/static/cloudcare/js/utils.js

```diff
@@ -380,7 +380,7 @@ hqDefine('cloudcare/js/utils', [
         let picker = hqTempusDominus.createDatePicker($el.get(0), options);
 
         $el.attr("placeholder", dateFormat);
-        $el.attr("pattern", "[0-9\-/]+");   // eslint-disable-line no-useless-escape
+        // $el.attr("pattern", "[0-9\-/]+");   // eslint-disable-line no-useless-escape
```

Comment:
What was this used for, and why comment out rather than delete?

Reply:
I think it is used for validation but it caused and error with each validation because it is malformed regex. I was not sure what the correct regex should be. So I commented it out for now.

Reply:
@esoergel do you have any context on what this is meant to be/do? I know you did work with the datepicker a while back and it looks like one of your commits put this pattern here.

Reply:
@nospame changed it to `$el.attr("pattern", "\\d{1,2}/\\d{1,2}/\\d{4}");` since the format is hard coded to 'M/D/YYYY'.

Reply:
I think this is meant to be "integers, dashes, or slashes" - IIRC, users were sometimes typing out "July 1st" or whatever, and this would mark that as an error.  The existing datepicker actually accepts string input in a variety of formats, defined here.  I believe it starts from left to right and uses the first format that matches:
https://github.com/dimagi/commcare-hq/blob/1a70089d74f4c7c367799ef83e7eddcd2739340a/corehq/apps/cloudcare/static/cloudcare/js/utils.js#L376-L376

Reply:
Some more context and discussion on that here: https://github.com/dimagi/commcare-hq/pull/32738

Reply:
that makes more sense. In the HTML it should end up looking like this `pattern="[0-9\-\/]+"` because we the dash and forward slash are inside the [] we need to escape them. But since we are in JS land we need two \ each to get there. so `$el.attr("pattern", "[0-9\\-\\/]+");`